server/utils/mpesa_utils.py

```
@@ -0,0 +1,278 @@
+        Check status of a transaction using Daraja API
+        """
+        try:
+            access_token = self.token_manager.get_token()
+            credentials = self.get_mpesa_credentials()
+            business_shortcode = credentials['business_shortcode']
+            passkey = credentials['passkey']
+            query_url = credentials['query_url']
+            timeout = credentials['timeout']
+
+            if not all([business_shortcode, passkey, query_url]):
+                raise Exception("MPESA query environment variables not configured")
```

Ruff (0.13.1)

Abstract `raise` to an inner function

(TRY301)